Tasks/Common/ServiceFabricHelpers/Get-ServiceFabricEncryptedText.ps1

```diff
@@ -12,7 +12,55 @@ function Get-ServiceFabricEncryptedText
 
     $defaultCertStoreName = "My"
     $defaultCertStoreLocation = "CurrentUser"
-    $serverCertThumbprints = $ClusterConnectionParameters["ServerCertThumbprint"]
+
+    if ($ClusterConnectionParameters["ServerCertThumbprint"])
+    {
+        $serverCertThumbprints = $ClusterConnectionParameters["ServerCertThumbprint"];
+    }
+    else 
+    {
+        try
+        {
+            $manifest = [xml](Get-ServiceFabricClusterManifest)
+            $securityInfo = $manifest.ClusterManifest.FabricSettings.Section | Where-Object {$_.Name -eq 'Security'}
+            $serverCertsParameter = $securityInfo.Parameter | Where-Object { $_.Name -eq 'ServerCertThumbprints' }
+            $serverCertThumbprints = $serverCertsParameter.Value.Split(',').trim()
+        }
+        catch 
+        {
+            if ($clusterConnectionParameters["ServerCommonName"])
+            {
+                $serverCertValues = $ClusterConnectionParameters["ServerCommonName"]
+                
+                # Get server cert thumbprints of valid certificates
+                if ($serverCertValues -is [array])
+                {
+                    foreach ($serverCertValue in $serverCertValues)
+                    {
+                        $serverCerts = Get-ChildItem -Path "Cert:\$defaultCertStoreLocation\$defaultCertStoreName" | Where-Object {$_.Subject -like $serverCertValue -and ($_.NotAfter -gt (Get-Date))}
+
+                        if ($serverCerts -is [array] -and $serverCerts.Length -gt 1) 
+                        {
+                            Write-Warning (Get-VstsLocString -Key MultipleCertPresentInLocalStoreWarningMsg -ArgumentList $serverCertValue)
+                            break
+                        }
+                        else 
+                        {
+                            $serverCertThumbprints = $serverCerts.Thumbprint
+                            if ($serverCertThumbprints)
+                            {
+                                break
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    $serverCertThumbprints = (Get-ChildItem -Path "Cert:\$defaultCertStoreLocation\$defaultCertStoreName" | Where-Object {$_.Subject -like $serverCertValues}).Thumbprint
```

Comment:
should we check for like bothways - `($_.Subject -like $serverCertValues) or ($serverCertValues -like $_.Subject)`
`like` only works one way

Reply:
Done